### PR TITLE
Dneto validation spv khr 16bit storage

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: source/val/validation_state.h [hunk 1]
```diff
@@ -359,8 +369,17 @@ class ValidationState_t {
   SpvAddressingModel addressing_model_;
   SpvMemoryModel memory_model_;
 
+  // Is OpTypeInt allowed with 16 bit width?
```
Comment (ehsannas) [inline source/val/validation_state.h]: `permit_int16_type_` and `permit_float16_type_` variables are not used. I think the `Feature` members are meant to do the work instead?

File: source/val/validation_state.h [hunk 2]
```diff
@@ -290,6 +296,10 @@ class ValidationState_t {
   bool IsStructTypeWithBuiltInMember(uint32_t id) const {
     return (builtin_structs_.find(id) != builtin_structs_.end());
   }
+
+  // Returns the state of optional features.
+  const Feature& features() { return features_; }
```
Comment (antiagainst) [inline source/val/validation_state.h]: This method can be declared as `const`.

File: source/val/validation_state.h (hunk 1 above)
Comment (dneto0) [inline source/val/validation_state.h]: Removed in next commit.  Thanks!

File: source/val/validation_state.cpp [hunk 3]
```diff
@@ -332,6 +332,24 @@ void ValidationState_t::RegisterCapability(SpvCapability cap) {
...
+      break;
+    case SpvCapabilityFloat16:
+    case SpvCapabilityFloat16Buffer:
+      features_.declare_float16_type = true;
+      break;
+    case SpvCapabilityStorageUniformBufferBlock16:
+    case SpvCapabilityStorageUniform16:
+    case SpvCapabilityStoragePushConstant16:
+    case SpvCapabilityStorageInputOutput16:
+      features_.declare_int16_type = true;
```
Comment (antiagainst) [inline source/val/validation_state.cpp]: Is this being too permissive? From my understanding, `SpvCapabilityInt16` enables more operations on int16 than `SpvCapabilityStorage*16`, but we don't actually distinguish that here. But we can handle this later maybe.

Comment (dneto0) [inline source/val/validation_state.cpp]: Correct, Int16 also allows arithmetic operations on those types.  But the validator doesn't check that yet.  Also, that's why the feature is called just *declare* 16bit types.   Later we will add something like int16_arithmetic, and that will distinguish the cases.

File: source/val/validation_state.h (hunk 2 above)
Comment (dneto0) [inline source/val/validation_state.h]: Done in next commit.

**Comment by dneto0** [comment]
All feedback addressed.

**Comment by dneto0** [comment]
Squashed and pushed into master as af7125dfb0

**Pull request closed without merging**